internal/pkg/oasparser/models/apiDefinition/openApi.go

```diff
@@ -34,72 +35,71 @@ func (swagger *MgwSwagger) SetInfoOpenApi(swagger3 openapi3.Swagger) {
-	swagger.resources = SetResourcesOpenApi3(swagger3)
+
+	swagger.vendorExtensible = converExtensibletoReadableFormat(swagger3.ExtensionProps)
+	swagger.resources = SetResourcesOpenApi(swagger3)
 
 	if IsServerUrlIsAvailable(swagger3) {
 		for i, _ := range swagger3.Servers {
 			endpoint := getHostandBasepathandPort(swagger3.Servers[i].URL)
-			swagger.productionUrls = append(swagger.productionUrls,endpoint)
+			swagger.productionUrls = append(swagger.productionUrls, endpoint)
 		}
-
 	}
 }
 
 func setOperationOpenApi(path string, pathtype string, operation *openapi3.Operation) Resource {
 	var resource Resource
-	resource = Resource{
-		path: path,
-		pathtype:   pathtype,
-		iD:      operation.OperationID,
-		summary: operation.Summary,
-		//Schemes: operation.,
-		tags: operation.Tags,
-		//Security: operation.Security.,
-		vendorExtensible: operation.Extensions}
+	if operation != nil {
+		resource = Resource{
+			path:     path,
+			pathtype: pathtype,
+			iD:       operation.OperationID,
+			summary:  operation.Summary,
+			description: operation.Description,
+			//Schemes: operation.,
+			//tags: operation.Tags,
+			//Security: operation.Security.,
+			vendorExtensible: converExtensibletoReadableFormat(operation.ExtensionProps)}
+	}
 	return resource
 }
 
-
-func SetResourcesOpenApi3(openApi openapi3.Swagger) []Resource {
+func SetResourcesOpenApi(openApi openapi3.Swagger) []Resource {
 	var resources []Resource
-
-	for path, pathItem := range openApi.Paths {
-		var resource Resource
-		if pathItem.Get != nil {
-			resource = setOperationOpenApi(path, "get", pathItem.Get)
-		} else if pathItem.Post != nil {
-			resource = setOperationOpenApi(path, "post", pathItem.Post)
-		} else if pathItem.Put != nil {
-			resource = setOperationOpenApi(path, "put", pathItem.Put)
-		} else if pathItem.Delete != nil {
-			resource = setOperationOpenApi(path, "delete", pathItem.Delete)
-		} else if pathItem.Head != nil {
-			resource = setOperationOpenApi(path, "head", pathItem.Head)
-		} else if pathItem.Patch != nil {
-			resource = setOperationOpenApi(path, "patch", pathItem.Patch)
-		} else {
-			//resource = setOperation(contxt,"get",pathItem.Get)
+	if openApi.Paths != nil {
+		for path, pathItem := range openApi.Paths {
+			var resource Resource
+			if pathItem.Get != nil {
+				resource = setOperationOpenApi(path, "get", pathItem.Get)
+			} else if pathItem.Post != nil {
+				resource = setOperationOpenApi(path, "post", pathItem.Post)
+			} else if pathItem.Put != nil {
+				resource = setOperationOpenApi(path, "put", pathItem.Put)
+			} else if pathItem.Delete != nil {
+				resource = setOperationOpenApi(path, "delete", pathItem.Delete)
+			} else if pathItem.Head != nil {
+				resource = setOperationOpenApi(path, "head", pathItem.Head)
+			} else if pathItem.Patch != nil {
+				resource = setOperationOpenApi(path, "patch", pathItem.Patch)
+			} else {
+				//resource = setOperation(contxt,"get",pathItem.Get)
+			}
```

Comment:
If the commented out line is not needed, please remove the empty `else` block

Reply:
This is already removed